notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,330 @@
+
+
+```python
+# HIDDEN
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from IPython.display import Image
+%matplotlib inline
+sns.set()
+sns.set_context('talk')
+```
+
+# Least Squares Regression as a Projection
+
+Recall that we found the optimal coefficients for linear models by optimizing their cost functions with gradient descent.
+
+We also mentioned that least squares linear regression can be solved analytically. While gradient descent is practical, this geometric perspective will provide a deeper understanding of linear regression.
+
+## Case Study
+
+We've been tasked with finding a good linear model and a rigorous argument for how we know that our model is sound, using the following data:
+
+| x | y |
+| - |:-|
+| 3 | 2 |
+| 0 | 1 |
+| -1 | -2 |
+
+
+```python
+# HIDDEN
+data = pd.DataFrame(
+    [
+        [3,2],
+        [0,1],
+        [-1,-2]
+    ],
+    columns=['x', 'y']
+)
+
+sns.regplot(x='x', y='y', data=data, ci=None)
```

Comment:
Add a `;` at the end of sns.regplot(...) in order to get rid of the printed output "<matplotlib.axes._subplots.AxesSubplot at 0x1111584e0>"

Reply:
excellent tip!